pkgs/development/python-modules/torch/default.nix

```diff
@@ -247,6 +248,9 @@ in buildPythonPackage rec {
   # Also avoids pytorch exporting the headers of pybind11
   USE_SYSTEM_PYBIND11 = true;
 
+  # NB technical debt: building without NNPACK as workaround for missing `six`
+  USE_NNPACK = 0;
```

Comment:
i noted this in https://github.com/NixOS/nixpkgs/issues/283343, but it looks like upstream stopped vendoring `six` and as a result `python_peachpy` fails to build (for some reason?) from some `third_party` submodule setup.

i don’t understand how nixpkgs manages this, but a workaround that incurs some different technical debt could be to patch the source so that it restores the expected vendored dep.

a better solution would be to understand how Nix actually supplies dependencies to submodules built this way, but i don’t know who best to ask about that.

Reply:
> What exactly are the implications? CC @samuela

unfortunately i'm not familiar with this aspect of pytorch's internals. it seems to me that we ought to be able to patch this though? it looks to me like something in the build process is overriding PYTHONPATH:
```
cd /build/source/build/confu-deps/NNPACK && PYTHONPATH=/build/source/build/confu-srcs/six:/build/source/third_party/python-peachpy /nix/store/w4fvvhkzb0ssv0fw5j34pw09f0qw84w8-python>
```
perhaps we can find where this is set in the build script?

Reply:
> What exactly are the implications? CC @samuela

I hadn't heard about NNPACK until trying to isolate the root cause of the build failure here. ["NNPACK is an acceleration package for neural network computations. NNPACK aims to provide high-performance implementations of convnet layers for multi-core CPUs."](https://github.com/Maratyszcza/NNPACK) Without it, PyTorch will fall back to using less efficient convnet algorithms, and it seems to be predominantly of use to mobile developers.

It would be nice to keep it in the build but think we could incur a bit of technical debt for now by building without it, focus on fixing the other build problems and get PyTorch 2.2.0 out the door to the Nix community with all its [fancy new features](https://github.com/pytorch/pytorch/releases/tag/v2.2.0).

@samuela yeah I reckon it's going to be in some submodule's `CMakeLists.txt ` and we could include a patch or something. Personally I just really want to use `torch` 2.2.0 and don't care about NNPACK.

Reply:
Agreed, I suspect NNPACK is not the headlining feature for most users, myself also included. I think it makes for this to not be a blocker to merging.

Reply:
> Agreed, I suspect NNPACK is not the headlining feature for most users, myself also included. I think it makes for this to not be a blocker to merging.

I agree. I will then mark this comment as resolved.

Reply:
- I'd be more wary of performance downgrades than of breakages, because breakages we do see, and benchmarks we don't run
- I'd be generally against the "fixes first, performance later" approach because Nixpkgs has been getting a misleading reputation for not ostensibly not optimizing their software/not being a serious option for hpc&c
- I'm leaning towards voting for the current solution, because I briefly looked into the NNPACK situation and it's cursed enough not to bother with it in this PR.
- I think fixing NNPACK would require rewriting the derivation so we can pass `cmakeFlags`, getting rid of the respective git-submodule, and packaging `NNPACK` separately (with a patch for `find_package`-compatibility).

Reply:
@samuela are you also alright with merging this without NNPACK ?
If so, I guess that you can do so. All other requests have been addressed.

Reply:
Tracking https://github.com/NixOS/nixpkgs/issues/288140

Reply:
> @samuela are you also alright with merging this without NNPACK ?

yup, fine with me